src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -105747,5 +105747,190 @@ class Program
+            comp.VerifyDiagnostics(
+                // (6,21): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object o2 = o1;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "o1").WithLocation(6, 21));
+
+            Assert.True(comp.RunNullableWalker);
+        }
+
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("annotations")]
+        [InlineData("warnings")]
+        [WorkItem(36131, "https://github.com/dotnet/roslyn/issues/36131")]
+        public void RunNullableWalker_GlobalDirective_WithNullDisables(string directive)
+        {
+            var source = $@"
+#nullable disable {directive}
+class C
+{{
+    void M(object? o1)
+    {{
+        object o2 = o1;
+    }}
+}}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(directive == "warnings" ?
+                DiagnosticDescription.None :
+                new[] {
+                    // (5,18): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
+                    //     void M(object? o1)
+                    Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(5, 18)
+                });
+
+            Assert.True(comp.RunNullableWalker);
+        }
+
+        [Fact]
+        [WorkItem(36131, "https://github.com/dotnet/roslyn/issues/36131")]
+        public void RunNullableWalker_GlobalDirectiveOff_WithNullableEnable()
+        {
+            RunNullableWalker_GlobalDirectiveOff_WithNullableDirective("",
+                // (7,21): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //         object o2 = o1;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "o1").WithLocation(7, 21));
+        }
+
+        [Fact]
+        [WorkItem(36131, "https://github.com/dotnet/roslyn/issues/36131")]
+        public void RunNullableWalker_GlobalDirectiveOff_WithNullableEnableAnnotations()
+        {
+            RunNullableWalker_GlobalDirectiveOff_WithNullableDirective("annotations");
+        }
+
+        [Fact]
+        [WorkItem(36131, "https://github.com/dotnet/roslyn/issues/36131")]
+        public void RunNullableWalker_GlobalDirectiveOff_WithNullableEnableWarnings()
+        {
+            RunNullableWalker_GlobalDirectiveOff_WithNullableDirective("warnings",
+                // (5,18): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context.
+                //     void M(object? o1)
+                Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(5, 18));
+        }
+
+        private void RunNullableWalker_GlobalDirectiveOff_WithNullableDirective(string directive, params DiagnosticDescription[] expectedDiagnostics)
```

Comment:
>private [](start = 8, length = 7)

consider 'static' here (I have a slightly easier time spotting out test helpers from tests when this modifier is present)